### PR TITLE
python3Packages.roundrobin: init at 0.0.4

### PR DESCRIPTION
**Pull request by wesnel**
###### Description of changes

This PR adds a new Python package, `roundrobin`. This package is useful because it [is used](https://github.com/locustio/locust/blob/3502aa610c89d6f07e21947f1e343819670d81eb/setup.cfg#L49) by the popular Python load testing tool `locust` (which I would also like to add to `nixpkgs` in the future). Thanks!

###### Things done

- Built on platform(s)
  - [ ] x86_64-linux
  - [ ] aarch64-linux
  - [x] x86_64-darwin
  - [ ] aarch64-darwin
- [x] For non-Linux: Is `sandbox = true` set in `nix.conf`? (See [Nix manual](https://nixos.org/manual/nix/stable/command-ref/conf-file.html))
- [x] Tested, as applicable:
  - [NixOS test(s)](https://nixos.org/manual/nixos/unstable/index.html#sec-nixos-tests) (look inside [nixos/tests](https://github.com/NixOS/nixpkgs/blob/master/nixos/tests))
  - and/or [package tests](https://nixos.org/manual/nixpkgs/unstable/#sec-package-tests)
  - or, for functions and "core" functionality, tests in [lib/tests](https://github.com/NixOS/nixpkgs/blob/master/lib/tests) or [pkgs/test](https://github.com/NixOS/nixpkgs/blob/master/pkgs/test)
  - made sure NixOS tests are [linked](https://nixos.org/manual/nixpkgs/unstable/#ssec-nixos-tests-linking) to the relevant packages
- [ ] Tested compilation of all packages that depend on this change using `nix-shell -p nixpkgs-review --run "nixpkgs-review rev HEAD"`. Note: all changes have to be committed, also see [nixpkgs-review usage](https://github.com/Mic92/nixpkgs-review#usage)
- [ ] Tested basic functionality of all binary files (usually in `./result/bin/`)
- [22.11 Release Notes (or backporting 22.05 Release notes)](https://github.com/NixOS/nixpkgs/blob/master/CONTRIBUTING.md#generating-2211-release-notes)
  - [ ] (Package updates) Added a release notes entry if the change is major or breaking
  - [ ] (Module updates) Added a release notes entry if the change is significant
  - [ ] (Module addition) Added a release notes entry if adding a new NixOS module
  - [ ] (Release notes changes) Ran `nixos/doc/manual/md-to-db.sh` to update generated release notes
- [x] Fits [CONTRIBUTING.md](https://github.com/NixOS/nixpkgs/blob/master/CONTRIBUTING.md).

### CODE REVIEW

File: pkgs/development/python-modules/roundrobin/default.nix [hunk 1]
```diff
@@ -0,0 +1,23 @@
...
+  pname = "roundrobin";
+  version = "0.0.2";
+
+  src = fetchFromGitHub {
+    owner = "linnik";
+    repo = pname;
+    rev = "5a5b34f2cd3ab8aa6d4135b31ab512b2d9338c19";
+    sha256 = "sha256-h+GcsHM0D4uBMcOng6mioC5o5AWbegiGqGYtuPoduM4=";
+  };
+
```
Comment (onny) [inline pkgs/development/python-modules/roundrobin/default.nix:16]: Can you add an additional pythonImportsCheco here?

**Review by fabaff — COMMENTED** [review]
There is a [test file](https://github.com/linnik/roundrobin/blob/master/test.py) in the repo. Could you check if the tests could be executed?

**Comment by wesnel** [comment]
> There is a [test file](https://github.com/linnik/roundrobin/blob/master/test.py) in the repo. Could you check if the tests could be executed?

@fabaff the tests do seem to get executed when i build it locally (see excerpt from my `stdout` below). if you think it would be preferable to add a custom `checkPhase`, though, i'd be happy to do that. i saw in some documentation that that may be best practices for python packages anyways.

```
➜  nix-build -A 'python39Packages.roundrobin' .                                                                                        ~/git/github.com/wesnel/nixpkgs
this derivation will be built:
  /nix/store/cycdlr6xywygxfhpi8vbbh8fzw86g6g2-python3.9-roundrobin-0.0.2.drv
building '/nix/store/cycdlr6xywygxfhpi8vbbh8fzw86g6g2-python3.9-roundrobin-0.0.2.drv'...
Sourcing python-remove-tests-dir-hook
Sourcing python-catch-conflicts-hook.sh
...
pythonImportsCheckPhase
Executing pythonImportsCheckPhase
setuptoolsCheckPhase
Executing setuptoolsCheckPhase
running test
WARNING: Testing via this command is deprecated and will be removed in a future version. Users looking for a generic test entry point independent of test runner are encouraged to use tox.
running egg_info
writing roundrobin.egg-info/PKG-INFO
writing dependency_links to roundrobin.egg-info/dependency_links.txt
writing top-level names to roundrobin.egg-info/top_level.txt
reading manifest file 'roundrobin.egg-info/SOURCES.txt'
reading manifest template 'MANIFEST.in'
warning: manifest_maker: MANIFEST.in, line 1: unknown action 'LICENSE'

adding license file 'LICENSE'
writing manifest file 'roundrobin.egg-info/SOURCES.txt'
running build_ext
test_basic (test.WeightCase) ... ok
test_smooth (test.WeightCase) ... ok
test_weighted (test.WeightCase) ... ok
test_weighted_not_smooth (test.WeightCase) ... ok

----------------------------------------------------------------------
Ran 4 tests in 0.001s

OK
Finished executing setuptoolsCheckPhase
/nix/store/g9hh5vqbwb584b0kxxrb1kx9zvisj278-python3.9-roundrobin-0.0.2
```

File: pkgs/development/python-modules/roundrobin/default.nix (hunk 1 above)
Comment (wesnel) [reply pkgs/development/python-modules/roundrobin/default.nix:16]: sure, good idea. i've added that now.

File: pkgs/development/python-modules/roundrobin/default.nix [hunk 2]
```diff
@@ -0,0 +1,30 @@
...
+  checkPhase = ''
+    ${python.interpreter} -m unittest discover
```
Comment (wesnel) [inline pkgs/development/python-modules/roundrobin/default.nix:19]: following up to my last comment: i learned in a different discussion about a way to do the `checkPhase` for python packages. i've just implemented that here.

**Comment by wesnel** [comment]
thanks @jonringer for inquiring about release tags in linnik/roundrobin#3. now that those have been added, i've updated this pull request to use the `0.0.4` tag.

**Comment by NickCao** [comment]
Upstream project no longer actively maintained.

**Pull request closed without merging**